### PR TITLE
Disable mission attempt button after clicking for the first time

### PR DESCRIPTION
**Pull request by allenwq**
Add a <code>data-disable-with</code> to prevent duplicate request at UI level.

\>< I found that the mission attempt/edit buttons are created by js, so here only js code change.

### CODE REVIEW

File: app/assets/javascripts/assignments.js [hunk 1]
```diff
@@ -25,7 +25,13 @@ $(document).ready(function() {
                             klass = 'btn-success';
                             break;
                     }
-                    $div.html('<a href="' + actions_map[mid].url + '" class="btn ' + klass + '" >' + actions_map[mid].action + '</a>')
+                    if ( actions_map[mid].action == 'Attempt' ){
```
Comment (allenwq) [inline app/assets/javascripts/assignments.js]: @jsyeo Do I need to fix this complains ? <code>actions_map</code> is an existing variable.

Comment (jsyeo) [inline app/assets/javascripts/assignments.js]: Better to fix the existing code smells if you happen to touch it. Can you fix it? Should be just a simple search and replace.

Comment (allenwq) [inline app/assets/javascripts/assignments.js]: I'm afraid that if I touched other lines, new complains will come and finally I need to fix the whole file ><

File: app/assets/javascripts/assignments.js [hunk 2]
```diff
@@ -25,7 +25,11 @@ $(document).ready(function() {
...
+                        $div.html('<a href="' + actions_map[mid].url + '" class="btn ' + klass + '" data-disable-with="Attempting..." >' + actions_map[mid].action + '</a>');
+                    } else {
+                        $div.html('<a href="' + actions_map[mid].url + '" class="btn ' + klass + '" >' + actions_map[mid].action + '</a>');
```
Comment (jsyeo) [inline app/assets/javascripts/assignments.js]: Can we do

    $('<a>', { href: 'actionsMap[mid.url]',
               class: 'brn ' + klass,
                   ... })

Comment (allenwq) [inline app/assets/javascripts/assignments.js]: :+1: I don't know this syntax

File: app/assets/javascripts/assignments.js (hunk 1 above)
Comment (jsyeo) [inline app/assets/javascripts/assignments.js]: :confused: okay lo.

**Comment by allenwq** [comment]
@jsyeo Based on your comments, I've updated here.

File: app/assets/javascripts/assignments.js [hunk 3]
```diff
@@ -25,7 +25,14 @@ $(document).ready(function() {
                             klass = 'btn-success';
                             break;
                     }
-                    $div.html('<a href="' + actions_map[mid].url + '" class="btn ' + klass + '" >' + actions_map[mid].action + '</a>')
+                    var link = $('<a></a>', {
```
Comment (jsyeo) [inline app/assets/javascripts/assignments.js]: Actually `$('<a>', ... )` will do

File: app/assets/javascripts/assignments.js [hunk 4]
```diff
@@ -25,7 +25,14 @@ $(document).ready(function() {
...
+                        'href': actions_map[mid].url,
```
Comment (jsyeo) [inline app/assets/javascripts/assignments.js]: `href: actions_map[mid].url,` will do. No need to quote.

Comment (allenwq) [inline app/assets/javascripts/assignments.js]: Thanks ! I've updated.

**Pull request merged**